src/runtime/gravity_core.c

```diff
@@ -3141,6 +3141,15 @@ static bool system_print (gravity_vm *vm, gravity_value_t *args, uint16_t nargs,
     return system_realprint(vm, args, nargs, rindex, true);
 }
 
+static bool system_input (gravity_vm *vm, gravity_value_t *args, uint16_t nargs, uint32_t rindex) {
+    #pragma unused(args,nargs)
+    char buffer[1024];
+    fgets(buffer, 1024, stdin);
+    // remove trailing newline captured by fgets
+    buffer[strlen(buffer) - 1] = 0;
```

Comment:
Note that here the newline if removed from the buffer, so remove this line of code if the terminator is needed.

Reply:
I improved that function with https://github.com/marcobambini/gravity/commit/3161cde05ac1735c9f6f7f5322373530c169337f